processor/resourcedetectionprocessor/README.md

```diff
@@ -665,9 +665,18 @@ resourcedetection:
         enabled: false
 ```
 
-## Ordering
+## Ordering and asynchronous detection
 
-Note that if multiple detectors are inserting the same attribute name, the first detector to insert wins. For example if you had `detectors: [eks, ec2]` then `cloud.platform` will be `aws_eks` instead of `ec2`. The below ordering is recommended.
+By default, the detectors are run asynchronously and therefore possibly out of order. If you want to keep the order of detectors, set the `async_detection` to `false`. This will also mean, that the retry mechanism for detectors will be disabled and therefore one pending detector can block the whole processor.
+
+```yaml
+processors:
+  resourcedetection:
+    detectors: [docker]
+    async_detection: false
```

Comment:
I wonder if we would want to introduce other configuration knobs related to detection/async. Is it worth introducting a new configuration section instead of a single flag?

Reply:
This depends on the fact if we are expecting some other configuration parameters to come in the future. If there is nothing in sight, I woulds say it's ok as it is, if there is something planned, let's change it

Reply:
If there is, potentially, new parameters that may come in the future, then I would want to do this, even if we have nothing specifically planned. Do you think people would potentially want to add other options here?

Reply:
currently all of the general parameters (`attributes`, `override` or http client settings) are part of the root config and these affect all of the detectors. I guess if we want to move it as a sub-parameter to a new `config` parameter, we should move all of these.

I would say it's ok to have in in the root next to the others

Reply:
removed the parameter